api/tests/routes/backoffice/accounts_test.py

```diff
@@ -2481,18 +2482,89 @@ def test_anonymize_public_account(
         [
             users_models.UserRole.PRO,
             users_models.UserRole.NON_ATTACHED_PRO,
-            users_models.UserRole.BENEFICIARY,
-            users_models.UserRole.UNDERAGE_BENEFICIARY,
+            users_models.UserRole.ANONYMIZED,
             users_models.UserRole.ADMIN,
         ],
     )
-    def test_anonymize_public_account_when_user_does_not_meet_criteria(
+    def test_anonymize_public_account_when_user_does_not_meet_roles_criteria(
         self,
-        legit_user,
         authenticated_client,
         role,
     ):
         user = users_factories.UserFactory(roles=[role])
 
         response = self.post_to_endpoint(authenticated_client, user_id=user.id)
         assert response.status_code == 400
+
+    @pytest.mark.parametrize(
+        "roles",
+        [
+            [users_models.UserRole.UNDERAGE_BENEFICIARY],
+            [users_models.UserRole.BENEFICIARY],
+            [],
+        ],
+    )
+    def test_anonymize_public_account_when_user_is_too_young(
+        self,
+        authenticated_client,
+        roles,
+    ):
+        user = users_factories.BeneficiaryFactory(
+            validatedBirthDate=datetime.datetime.today() - datetime.timedelta(days=365 * 21), roles=roles
+        )
+
+        response = self.post_to_endpoint(authenticated_client, user_id=user.id)
+        assert response.status_code == 400
+
+    @pytest.mark.parametrize(
+        "roles",
+        [
+            [users_models.UserRole.UNDERAGE_BENEFICIARY],
+            [users_models.UserRole.BENEFICIARY],
+            [],
+        ],
+    )
+    def test_anonymize_public_account_when_user_has_no_deposits(
+        self,
+        authenticated_client,
+        roles,
+    ):
+        user = users_factories.BeneficiaryFactory(
+            roles=roles,
+        )
+
+        finance_models.Deposit.query.filter(finance_models.Deposit.user == user).delete()
+
+        response = self.post_to_endpoint(authenticated_client, user_id=user.id)
+        assert response.status_code == 302
+
+        expected_url = url_for("backoffice_web.public_accounts.get_public_account", user_id=user.id, _external=True)
+        assert response.location == expected_url
+
+        assert user.roles == [users_models.UserRole.ANONYMIZED]
+
+    @pytest.mark.parametrize(
+        "roles",
+        [
+            [users_models.UserRole.UNDERAGE_BENEFICIARY],
+            [users_models.UserRole.BENEFICIARY],
+            [],
+        ],
```

Comment:
Proposition pour plus de cohérence : 
```suggestion
        "factory",
        [
            [users_factories.BeneficiaryGrant18Factory],
            [users_factories.UnderageBeneficiaryFactory],
            [users_factories.UserFactory],
        ],
```
et adapter en-dessous.